### PR TITLE
Make intra cluster tls work

### PR DESCRIPTION
**Pull request by michaeljmarshall**
## Motivation

Configure all components to use TLS when networking with each other.

## Modifications

- Update several config maps to enable hostname verification.
- Update the Pulsar cluster initialization script so that the function worker discovers the correct zookeeper host and port for TLS
- Add a zero-trust example values file as an example of how to configure Pulsar for a complete TLS deployment

## Notes
One important warning is that that the TLS deployment will only correctly work when deploying the brokers as a Stateful Set. I'm not sure that this requirement was explicitly stated before. However, it is essential in order to ensure that brokers are addressable by a unique DNS name because their `advertisedAddress` is used in the Pulsar protocol, and clients will fail connections due to hostname verification failure if the broker's advertised address is not a DNS entry pointing to that broker.

### CODE REVIEW

File: helm-chart-sources/pulsar/templates/broker-sts/broker-sts-statefulset.yaml [hunk 1]
```diff
@@ -283,16 +283,12 @@ spec:
         envFrom:
         - configMapRef:
             name: "{{ template "pulsar.fullname" . }}-{{ .Values.brokerSts.component }}"
-        {{- if .Values.brokerSts.ledger }}
         env:
-        - name: advertisedAddress
-          valueFrom:
-            fieldRef:
-              fieldPath: status.podIP
```
Comment (michaeljmarshall) [inline helm-chart-sources/pulsar/templates/broker-sts/broker-sts-statefulset.yaml:291]: @eolivelli - in order to make TLS hostname verification work within the k8s cluster, I removed this block that sets `advertisedAddress` to an IP address. The primary reason is that when a broker returns an owning broker for a lookup request, it uses the advertised address, and the client will verify the hostname based on the advertised address. Do you see any issues with making that change? When the broker is a stateful set, the resulting `advertisedAddress` appears to be `pulsar-broker-0.pulsar-broker.default.svc.cluster.local`. That is still a unique identifier that will allow extensions like KoP to make network calls to other brokers.

Comment (eolivelli) [reply helm-chart-sources/pulsar/templates/broker-sts/broker-sts-statefulset.yaml:291]: we are good as far as advertisedAddress points to a address reachable by other brokers and by proxy

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by michaeljmarshall** [comment]
Closing to split into multiple PRs.

**Pull request closed without merging**